Improve readme and CDT Cloud update logo

planger opened the pull request:
#### What it does
This change refocuses the README to better resonate with potential users by presenting the extension's features and usage more clearly. We aim to maintain essential information for contributors, ensuring a balance between user orientation and developer guidance.

Also we update the CDT Cloud logo, fixed broken links, and adapted wrong information in the CONTRIBUTING guide left over from Theia.

#### How to test
Note that the link to the screenshot only works after this is merged. Otherwise we would need to use a relative link, which may not work when displayed in the marketplace or on open-vsx.org.

#### Review checklist

- [X] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

arekzaluski commented on README.md:
```diff
@@ -1,27 +1,50 @@
 # Memory Inspector
 
-This extension contributes a set of widgets for viewing memory in different ways.
+A Visual Studio Code extension that provides a powerful and configurable memory viewer that works with debug adapters.
```
👍

planger commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -13,8 +13,8 @@ Contributors guide: https://github.com/eclipse-cdt-cloud/vscode-memory-inspector
 #### Reminder for reviewers
 
-- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
+- as a reviewer, I agree to behave in accordance with [the code of conduct](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
```
Darn, I meant to point to the CoC of this repo. Thanks for catching that!

planger commented on README.md:
```diff
@@ -1,27 +1,50 @@
+- **Multiple Format Display**: Shows memory data in different formats, provided your debug adapter supports the `ReadMemoryRequest`.
+- **Address Navigation**: Easily jump to and scroll through memory addresses.
+- **Variable Highlights**: Marks variable memory ranges.
+- **Edit Memory**: Allows in-place memory editing, if the debug adapter supports the `WriteMemoryRequest`.
```
Yeah, you are right. I'm an optimist, but I'm completely fine to leave this PR open until #108 is merged.

jreineckearm replied:
I am sure it will come for the next release one or the other way. ;-)

planger commented:
Thank you for the fast review @colin-grant-work! I accepted your suggestion and fixed the link. Happy to leave this open until #108 is merged to not promise something before it is actually merged :-)

jreineckearm commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -13,8 +13,8 @@ Contributors guide: https://github.com/eclipse-cdt-cloud/vscode-memory-inspector
 #### Review checklist
 
-- [ ] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)
+- [ ] as an author, I have thoroughly tested my changes and carefully followed [the contribution guidelines](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/blob/main/CONTRIBUTING.md)
```
👍

jreineckearm commented on README.md:
```diff
@@ -1,27 +1,50 @@
+## Features
 
-## The Memory Provider
+- **Multiple Format Display**: Shows memory data in different formats, provided your debug adapter supports the `ReadMemoryRequest`.
```
`Multiple Format`: I don't think we have that yet in the data contents area. Maybe name it `Configurable Memory Display` and mention the `Multiple Format Display` for hover-over?

jreineckearm commented on README.md:
```diff
@@ -1,27 +1,50 @@
-## The Memory Provider
+- **Multiple Format Display**: Shows memory data in different formats, provided your debug adapter supports the `ReadMemoryRequest`.
+- **Address Navigation**: Easily jump to and scroll through memory addresses.
+- **Variable Highlights**: Marks variable memory ranges.
```
```suggestion
- **Variable Highlights**: Colors memory ranges for variables.
```

jreineckearm commented on README.md:
```diff
@@ -1,27 +1,50 @@
+- **Address Navigation**: Easily jump to and scroll through memory addresses.
+- **Variable Highlights**: Marks variable memory ranges.
+- **Edit Memory**: Allows in-place memory editing, if the debug adapter supports the `WriteMemoryRequest`.
+- **Memory Management**: Enables saving and restoring memory data for specific address ranges.
```
Should we explicitly mention the file format?

jreineckearm commented on README.md:
```diff
@@ -1,27 +1,50 @@
+- **Variable Highlights**: Marks variable memory ranges.
+- **Edit Memory**: Allows in-place memory editing, if the debug adapter supports the `WriteMemoryRequest`.
+- **Memory Management**: Enables saving and restoring memory data for specific address ranges.
+- **Custom Views**: Create and customize as many memory views as you need.
```
```suggestion
- **Customized Views**: Create and customize as many memory views as you need.
```

jreineckearm commented on README.md:
```diff
@@ -1,27 +1,50 @@
+## Getting Started
+
+1. **Install**: Add the extension to VS Code.
+2. **Debug Session**: Start with a debug adapter that has the [`ReadMemory` request](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_ReadMemory) capability.
```
We should point users to the global `Debug Types` setting to enable this for any adapter that supports the `ReadMemory` request.

![image](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/assets/24247651/459016a1-7718-46c2-8aca-b9fc095727a6)

jreineckearm reviewed (commented):
Looks good! A couple of comments.

planger commented:
@jreineckearm Thank you for your feedback! I think I addressed all your comments and accepted all suggestions.

jreineckearm commented:
Thanks for the updates! Looks good to me!

planger commented:
Thanks everyone for their review and suggestions! @colin-grant-work Can you please take another look and, if you're fine with this change, give your approval? Thank you!
I'll wait for https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/pull/108 to be merged before I merge this change.

jreineckearm commented:
@planger , @colin-grant-work , as discussed today we should add a terminology section and a screenshot to highlight some of the terminology usage. See https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/issues/100#issuecomment-2008003753 for a summary of that call.

I'd recommend to use this PR for those editions. I am happy to draft something and add it here. But it's getting late now. I'll try to provide a write up tomorrow.

jreineckearm commented:
Apologies for taking longer than expected. The next comment is a proposal for the wording between "Configuration" and "Contributing". I'll prepare a screenshot next.

jreineckearm commented:
## Memory Format Settings

The Memory Format settings allow to configure how data that is read from the target system is interpreted and displayed.
Use the following to retarget the view to your needs and the inspected memory architecture:
1. **Bytes per MAU**: The number of _Bytes_ that form the Minimum Addressable Unit. It commonly is fixed number for a specific target hardware. Use for example a value of `1` for byte-addressable architectures.
2. **MAUs per Group**: The number of _MAUs_ that form a _Group_ considering the selected _Endianess_. Use for example a value of `2` to form a 4-byte value consisting of `2` 2-byte MAUs.
3. **Groups per Row**: Number of _Groups_ to display in a row. This can be a fixed number of _Groups_. Or the value `Autofit` to let the Memory Inspector calculate the best utilization of space in the `Data` column.
4. **Group Endianess**: The order of _MAUs_ within a _Group_. The value can be `Little Endian` or `Big Endian`.


The following terminology is used:
- **Byte**: A data unit of 8 Bits.
- **MAU (Minimum Addressable Unit)**: A Minimum Addressable Unit of memory. It consists of one or more _Bytes_ represented by a single address.
- **Group**: A group of _MAUs_ that forms a data value. _Groups_ are the Memory Inspector's default granularity to edit memory contents.
- **Row**: A row in the Memory Inspector display containing multiple _Groups_.
- **Endianess**: The order of displaying _MAUs_ within a _Group_.

planger commented:
Thanks @jreineckearm! I've added your section to the readme and also updated the screenshot to how it'll likely look like after renaming *words* to *MAUs*.

jreineckearm commented:
Thanks @planger , may need to do another review iteration of the text and formatting. Already spotted at least one typo in the text I provided. :-)
The link to the screenshot seems to be broken on the branch. Gives me a 404 when I click it.

planger commented:
> The link to the screenshot seems to be broken on the branch. Gives me a 404 when I click it.

Yes, this link only works after we've merged this change as the image then should become available under the given URL. I preferred to use absolute URLs over relative ones, as this should be safer than something relative when showing the screenshot in the readme of the published extension on the VSX registries.

jreineckearm commented on README.md:
```diff
@@ -1,27 +1,76 @@
+The Memory Format settings allow to configure how data that is read from the target system is interpreted and displayed.
+Use the following to retarget the view to your needs and the inspected memory architecture:
+
+1. **Bytes per MAU**: The number of *Bytes* that form the Minimum Addressable Unit. It commonly is fixed number for a specific target hardware. Use for example a value of `1` for byte-addressable architectures.
```
```suggestion
1. **Bytes per MAU**: The number of *Bytes* that form the Minimum Addressable Unit. It commonly is a fixed number for a specific target hardware. Use for example a value of `1` for byte-addressable architectures.
```

jreineckearm reviewed (commented):
Found a tiny typo. Waiting for #111 before taking a screenshot to highlight the Memory Format option impact.

jreineckearm commented:
![image](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/assets/24247651/50d47fa4-7925-4414-8e25-c90b24e619ec)

@planger , could you please let me know if attached (modified) screenshot works as a visual reference for the various options?
Unfortunately, a little harder to explain endianess in a single one and wouldn't want to put too many screenshots into one sections.

planger commented:
@jreineckearm I like it! Thank you!
Just two suggestions for minor tweaks:
1. It could help to not use *Autofit* in this screenshot but set it to a concrete value (e.g. 2). This would make it easy to understand, given that there would then be the same number of groups in a row (e.g. 2).
2. I'm myself not sure about this one, but we could put an example in the text (enumeration in the readme) for the *Group Endianess* setting. This example could match a specific group in the screenshot to show how the row would be like with Little Endian and Big Endian.

jreineckearm commented:
Thanks for the feedback, @planger !
Agree with both items. Hope those two images work better. :-)

![image](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/assets/24247651/f50634d0-c3ff-49fa-adaa-74481fba8809)

![image](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/assets/24247651/67ced9b5-8489-4247-bf8a-b9bffe4258d9)

jreineckearm commented:
And another one with the item number. But probably a bit too much.

![image](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/assets/24247651/72e44f02-c022-4152-83a0-f7369517ace3)

jreineckearm reviewed (approved):
Thanks! Looks good to me. :-)

planger commented:
@jreineckearm Thank you for the images! Imho they look great. I've now used the one with the number regarding endianness. Imho it is more consistent, but I don't have a strong opinion. I cropped the first image a bit to better align both screenshots and avoid a light border.

As you can't preview it on Github because of the absolute link, I put a screenshot below:

![image](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/assets/588090/5e379925-fad2-4076-8bf4-9417b94affdc)

Please let me know what you think!

jreineckearm commented:
> I cropped the first image a bit to better align both screenshots and avoid a light border.

Thanks! I must have missed that border. IMO good to go with that.

planger commented:
Thanks @jreineckearm!

colin-grant-work commented:
My approval stands - merge at your convenience.
